bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java

```diff
@@ -224,7 +225,7 @@ public void testEstimatedAndSplitForSmallTable() throws Exception {
         CloudBigtableScanConfiguration config =
             new CloudBigtableScanConfiguration(projectId, zoneId, clusterId,
                 tableName.getQualifierAsString(), new Scan());
-        CloudBigtableIO.Source source = new CloudBigtableIO.Source(config);
+        CloudBigtableIO.Source source = (Source) CloudBigtableIO.read(config);
```

Comment:
This does indeed sound like a code smell.  The idea of CloudBigtableIO.read() should just be to get the generic Dataflow object, but we're reaching under the covers here.

We're actually trying to test the sample row key processing, and need to reach under the covers to get to it.  I should really move the logic and this test to separate classes.  I'll add a TODO for this PR.